### PR TITLE
added readme for link unfurling scenario

### PR DESCRIPTION
**Pull request by tsuwandy**

### CODE REVIEW

**Comment by fuselabs** [comment]
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Builder/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.AI.Luis.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Builder.AI.Luis/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.AI.QnA.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Builder.AI.QnA/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.ApplicationInsights.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Builder.ApplicationInsights/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.Azure.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Builder.Azure/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.Dialogs.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Builder.Dialogs/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.TemplateManager.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Builder.TemplateManager/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Configuration.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Configuration/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Connector.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Connector/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Schema.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Schema/4.3.1)

File: tests/Teams/LinkUnfurling/README.md [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+  - Navigate to tests\Teams folder and click on Scenarios.sln
+  - Set LinkUnfurling project as StartUp Project
+  - Open launchSettings.json under Properties and update applicationUrl fields to http://localhost:3978/
+  - Press `F5` to run the project
+
+### Interacting with the bot
+
+1. Send a message to your bot in Teams
+2. Type any valid url, ie http://www.bing.com, wait until the url is bolded, hit the space key and you should see a thumbnail card for the url info populated, like below.
+   ![1569017810114](D:\src\botbuilder-dotnet\tests\Teams\LinkUnfurling\1569017810114.png)
```
Comment (daveta) [inline tests/Teams/LinkUnfurling/README.md]: >D:\src\botbuilder-dotnet\tests\Teams\LinkU [](start = 20, length = 42)

Update to relative

File: tests/Teams/LinkUnfurling/README.md [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+
+3.  Url already typed in in earlier chat is cached and you might not see the thumbnail if you enter the same url.  Try step #2 with different url.
+
+### Connect to the bot using Bot Framework Emulator
+
+- Launch Bot Framework Emulator
+- File -> Open Bot
+- Enter a Bot URL of `http://localhost:3978/api/messages`
```
Comment (daveta) [inline tests/Teams/LinkUnfurling/README.md]: Remove

File: tests/Teams/LinkUnfurling/README.md (hunk 1 above)
Comment (tsuwandy) [reply tests/Teams/LinkUnfurling/README.md]: Done!

File: tests/Teams/LinkUnfurling/README.md (hunk 2 above)
Comment (tsuwandy) [reply tests/Teams/LinkUnfurling/README.md]: Done!

**Comment by fuselabs** [comment]
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Builder/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.AI.Luis.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Builder.AI.Luis/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.AI.QnA.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Builder.AI.QnA/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.ApplicationInsights.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Builder.ApplicationInsights/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.Azure.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Builder.Azure/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.Dialogs.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Builder.Dialogs/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.TemplateManager.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Builder.TemplateManager/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Configuration.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Configuration/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Connector.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Connector/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Schema.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Schema/4.3.1)

File: tests/Teams/LinkUnfurling/TeamsAppManifest/manifest.json [hunk 3]
```diff
@@ -54,7 +54,7 @@
           "type": "link",
           "value": {
             "domains": [
-              "tzadik.com"
+              "*.com"
```
Comment (johnataylor) [inline tests/Teams/LinkUnfurling/TeamsAppManifest/manifest.json]: can we have a site that's not everything? for example, "microsoft.com" and something else...

File: tests/Teams/LinkUnfurling/README.md [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+  - Open launchSettings.json under Properties and update applicationUrl fields to http://localhost:3978/
+  - Press `F5` to run the project
+
+### Interacting with the bot
+
+1. Send a message to your bot in Teams
+2. Type any valid url, ie http://www.bing.com, wait until the url is bolded, hit the space key and you should see a thumbnail card for the url info populated, like below.
+   ![1569017810114](1569017810114.png)
+
+3.  Url already typed in in earlier chat is cached and you might not see the thumbnail if you enter the same url.  Try step #2 with different url.
```
Comment (stevengum) [inline tests/Teams/LinkUnfurling/README.md]: One thing that @clearab mentioned is that if you have multiple Bots/Apps installed that perform Link Unfurling, whomever is fastest wins.

Also, if a bot returns more than one search result for an unfurled link, only the first result is shown to the user. ([docs](https://docs.microsoft.com/en-us/microsoftteams/platform/concepts/messaging-extensions/search-extensions#receive-requests-from-links-inserted-into-the-compose-message-box))

**Comment by fuselabs** [comment]
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Builder/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.AI.Luis.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Builder.AI.Luis/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.AI.QnA.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Builder.AI.QnA/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.ApplicationInsights.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Builder.ApplicationInsights/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.Azure.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Builder.Azure/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.Dialogs.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Builder.Dialogs/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.TemplateManager.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Builder.TemplateManager/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Configuration.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Configuration/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Connector.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Connector/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Schema.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Schema/4.3.1)

File: tests/Teams/LinkUnfurling/README.md [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+  - File -> Open -> Project/Solution 
+  - Navigate to tests\Teams folder and click on Scenarios.sln
+  - Set LinkUnfurling project as StartUp Project
+  - Open launchSettings.json under Properties and update applicationUrl fields to http://localhost:3978/
+  - Press `F5` to run the project
+
+### Interacting with the bot
+
+1. Send a message to your bot in Teams
+2. Type any valid url, ie http://www.bing.com, wait until the url is bolded, hit the space key and you should see a thumbnail card for the url info populated, like below.
```
Comment (EricDahlvang) [inline tests/Teams/LinkUnfurling/README.md]: Should this be  .microsoft.com or one of the addresses in the manifest? (Or *.bing.com could be in the manifest as well)

Comment (tsuwandy) [reply tests/Teams/LinkUnfurling/README.md]: good catch, let me just add bing.com in there. :-)

**Comment by fuselabs** [comment]
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Builder/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.AI.Luis.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Builder.AI.Luis/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.AI.QnA.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Builder.AI.QnA/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.ApplicationInsights.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Builder.ApplicationInsights/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.Azure.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Builder.Azure/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.Dialogs.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Builder.Dialogs/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.TemplateManager.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Builder.TemplateManager/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Configuration.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Configuration/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Connector.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Connector/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Schema.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Schema/4.3.1)

**Pull request merged**